patroni/postgresql.py

```diff
@@ -557,7 +561,8 @@ def _custom_bootstrap(self, config):
         if 'recovery_conf' in config:
             self.write_recovery_conf(config['recovery_conf'])
         elif os.path.isfile(self._recovery_conf) or os.path.islink(self._recovery_conf):
-            os.unlink(self._recovery_conf)
+            if not keep_existing_recovery_conf:
```

Comment:
I think it could be simplified to:
```python
        elif (os.path.isfile(self._recovery_conf) or os.path.islink(self._recovery_conf)) and \
                not config.get('keep_existing_recovery_conf'):
            os.unlink(self._recovery_conf)
```

And lines 547-550 could be dropped.